patroni/api.py

```diff
@@ -415,6 +415,11 @@ def query(self, sql, *params, **kwargs):
 
     def get_postgresql_status(self, retry=False):
         try:
+            cluster = (
+                self.server.patroni.dcs.cluster or
+                self.server.patroni.dcs.get_cluster()
```

Comment:
I would actually avoid calling `get_cluster` method and always use the cached version.

Also the line https://github.com/zalando/patroni/pull/764/files#diff-790530624eaa21e5fb614fc321aef277R460 could be removed.